testbenches/common/v/profiler.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+namespace bsg_profiler {
+class profiler {
+public:
+    // constructors
+    profiler()
+        : is_exit(0)
+        , trace_file(-1) {
+    }
+
+    // members
+    int is_exit;
+    int trace_file;
```

Comment:
Are we assuming these variables are atomically updated because the simulation is single-threaded?

Reply:
I guess that is an assumption I made; I didn't make any of these function re-entrant.

Reply:
Dan might have an interesting input here, having worked with Verilator.

Reply:
making the init and exit functions thread safe shouldn't be too much work

Reply:
@dpetrisko are you using parallel simulation?

Reply:
k. i made these functions thread safe. it doesn't sound like we have any parallel simulation but just in case we ever do, that's one less difficult bug to track down.

Reply:
I think it's fine as long as there is no significant performance overhead, but it seems like we don't actually know if there is a simulator where this mutex is required.

Reply:
Maybe it's possible to have one log file for each tile, then you kinda avoid all these problems.

Reply:
I think there's value to having it be one trace file. It makes the scripts that read this in simpler, and changing it would require rewriting those scripts. 

Whatever overhead from the mutex exists at init and exit time is amortized over the life of simulation so I think the current solution is quite workable.